model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/solve_nonhydro.py

```diff
@@ -436,127 +436,201 @@ def __init__(
+            backend=self._backend,
+        )
+
+        self._compute_theta_and_exner = compute_theta_and_exner.with_backend(self._backend).compile(
+            offset_provider_type={}
+        )
+        self._compute_exner_from_rhotheta = compute_exner_from_rhotheta.with_backend(
+            self._backend
+        ).compile(offset_provider_type={})
+        self._update_theta_v = update_theta_v.with_backend(self._backend).compile(
+            offset_provider_type={}
+        )
         self._init_two_cell_kdim_fields_with_zero_vp = (
-            init_two_cell_kdim_fields_with_zero_vp.with_backend(self._backend)
+            init_two_cell_kdim_fields_with_zero_vp.with_backend(self._backend).compile(
+                offset_provider_type={}
+            )
         )
         self._compute_approx_of_2nd_vertical_derivative_of_exner = (
-            compute_approx_of_2nd_vertical_derivative_of_exner.with_backend(self._backend)
+            compute_approx_of_2nd_vertical_derivative_of_exner.with_backend(self._backend).compile(
+                offset_provider_type=self._grid.offset_providers
+            )
         )
         self._compute_perturbation_of_rho_and_theta = (
-            compute_perturbation_of_rho_and_theta.with_backend(self._backend)
+            compute_perturbation_of_rho_and_theta.with_backend(self._backend).compile(
+                offset_provider_type={}
+            )
         )
         self._mo_icon_interpolation_scalar_cells2verts_scalar_ri_dsl = (
-            mo_icon_interpolation_scalar_cells2verts_scalar_ri_dsl.with_backend(self._backend)
+            mo_icon_interpolation_scalar_cells2verts_scalar_ri_dsl.with_backend(
+                self._backend
+            ).compile(offset_provider_type=self._grid.offset_providers)
         )
         self._init_two_edge_kdim_fields_with_zero_wp = (
-            init_two_edge_kdim_fields_with_zero_wp.with_backend(self._backend)
+            init_two_edge_kdim_fields_with_zero_wp.with_backend(self._backend).compile(
+                offset_provider_type={}
+            )
         )
         self._compute_hydrostatic_correction_term = (
-            compute_hydrostatic_correction_term.with_backend(self._backend)
+            compute_hydrostatic_correction_term.with_backend(self._backend).compile(
+                offset_provider_type=self._grid.offset_providers
+            )
         )
         self._compute_theta_rho_face_values_and_pressure_gradient_and_update_vn = compute_edge_diagnostics_for_dycore_and_update_vn.compute_theta_rho_face_values_and_pressure_gradient_and_update_vn.with_backend(
             self._backend
+        ).compile(
+            #cpd=[constants.CPD],
+            #iau_wgt_dyn=[self._config.iau_wgt_dyn],
+            #is_iau_active=[self._config.is_iau_active],
+            #limited_area=[self._grid.limited_area],
+            #iadv_rhotheta=[self._config.iadv_rhotheta],
+            #igradp_method=[self._config.igradp_method],
```

Comment:
if all of these are enabled, then gtfn will error in some transformation (for the mch_dsl experiment), the current selection is a random set, not the minimal set...